src/union.ts

```diff
@@ -163,7 +165,7 @@ export class Union {
         }
 
         let lastError: IUnionFsError | null = null;
-        let result: Set<string> = new Set();
+        let result: Map<string, readdirEntry> = new Map();
```

Comment:
Generally it's nicer to have TypeScript infer types in places like this, as it moves the focus from "can x be assigned to y?" to "can x be used like this?"

Reply:
So, how should this look instead?

Reply:
`let result = new Map<string, readdirEntry>();`